### ISSUE TITLE
Role.addManagedPolicy does not work for imported roles

### ISSUE THREAD

**Issue by its-mirus-lu** [aws/aws-cdk#8307]
### The Scenario
TLDR; aws_iam.Role.attachManagedPolicy does not attach the specified managed policy to the role, aws_iam.ManagedPolicy.attachToRole does.

I have two stacks: one stack deploys roles another deploys an application stack (both are in the same account).  The roles stack is deployed first.

In the application stack, I would like to attach a managed policy to a role in the first.

Attempting to use aws_iam.Role.attachManagedPolicy does not create the association but aws_iam.ManagedPolicy.attachToRole does.

### Environment

  - **CDK CLI Version:** 1.39.0 (build 5d727c1) 
  - **Module Version:** 1.38.0
  - **Node.js Version:** v14.0.0 
  - **OS:** macOS Mojave 10.14.6 (18G4032) 
  - **Language:** Typescript and Python 

#### Steps to reproduce
1. Deploy one stack with an IAM role
1. In a second stack create a managed policy
1. In the second stack Import the role using aws_iam.Role.fromRoleArn (importedRole)
1. In the second stack attempt to add the managed policy to importedRole via importedRole.attachManagedPolicy

##### Observed
1. cdk synth does not show any associations made between the role and the managed policy
1. using ManagedPolicy.attachToRole works however

##### Expected
1. cdk synth (and subsequently) cdk deploy should associate the role and managed policy when Role.attachManagedPolicy is used

**Comment by miruslu-amazon** [comment]
It looks like addManagedPolicy is not implemented... https://github.com/aws/aws-cdk/blob/v1.42.1/packages/@aws-cdk/aws-iam/lib/role.ts#L216

Is there a way that this can be __clearly__ stated in documentation?

**Comment by rix0rrr** [comment]
We can certainly add that to the documentation.

**Comment by Chriscbr** [comment]
Could this be resolved by changing addManagedPolicy to take a `ManagedPolicy` instead of `IManagedPolicy` as its argument (and then calling the appropriate method on the managed policy to attach it to the role)? 

I feel like this could fit in with the design guidelines - "It's okay to enable multiple ways to achieve the same thing, in order to make it more natural for users who come from different mental models." Also the `attachInlinePolicy` method on imported roles requires a `Policy` (opposed to an `IPolicy`), so this wouldn't be especially out of place.

**Comment by OperationalFallacy** [comment]
Just run into this bug, trying to configure some policies for roles managed outside of cdk app 😢

**Comment by guilarteH** [comment]
Same here, this is really unfortunate. It fails silently and doesn't attach anything.
I guess I'll be forced to create a proper inline policy and do the right thing following the principle of least privilege.

**Comment by valters** [comment]
We just hit this issue. Especially surprising that even a warning is not getting printed. https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-iam/lib/role.ts#L227

**Comment by jayhilden** [comment]
The workaround I found is below.  It's not perfect, but it's pretty good:

1. define the policy statement(s)
1. create the managed policy with that policy statement
1. attach the policy statement to the imported role.

## Example
```
const statement = new PolicyStatement({
    effect: Effect.ALLOW,
    actions: ['execute-api:Invoke'],
    resources: [
        //SOME RESOURCE
    ]
})
this.cloudAuthAccessPolicy = new ManagedPolicy(this, 'FOO', {
    managedPolicyName: 'FOO',
    description: "BAR",
    statements: [
        statement
    ]
})

const role = Role.fromRoleArn(this, `name`, `arn:aws:iam::${props.account}:role/SOME_ROLE_NAME`)
role.addToPolicy(statement)
```

**Comment by digven** [comment]
I know this issue has been open for some time now. Is there a fix team is working on ?. I have a use case where I want to create roles and policies separately and then associate them later using Role.addManagedPolicy. 

The only way right now is to associate the role to policy during the policy creation as mentioned in above comment which is not ideal.

**Comment by rdinardi-bw** [comment]
This is still wrong in the [documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.FromRoleArnOptions.html#mutable) when importing an existing role. 

One can set a `mutable` option when importing the role (by default it's set to `true`) The docs say:
>  Whether the imported role can be modified by attaching policy resources to it.

Which is untrue. You may only modify the role by adding inline policies. Not managed policies.

**Comment by Lanayx** [comment]
This shouldn't be marked as documentation issue, it rather should just be implemented - I don't see anything wrong in idea of adding managed policies to the imported role (for example to the auto-generated service roles).

**Comment by swachter** [comment]
As a workaround, you can use a AwsCustomResource to attach a managed policy to a role:

```
new AwsCustomResource(this, 'developerReadOnlyAccess', {
    onCreate: {
        service: 'IAM',
        action: 'attachRolePolicy',
        parameters: {
            PolicyArn: 'arn:aws:iam::aws:policy/ReadOnlyAccess',
            RoleName: 'developer'
        },
        physicalResourceId: PhysicalResourceId.of(`${this.stackName}-dev-read-only-access`)
    },
    policy: AwsCustomResourcePolicy.fromSdkCalls({ resources: AwsCustomResourcePolicy.ANY_RESOURCE })
});
```
